### ISSUE TITLE
Occasional Mueller Mandel SN prescription error

### ISSUE THREAD

**Issue by SimonStevenson** [TeamCOMPAS/COMPAS#1218]
**Describe the bug**
When running large populations, I occasionally (rarely, something like 1 in 10^5 binaries) get the following error:

ERROR:  in Early_Asymptotic_Giant_Branch::double GiantBranch::CalculateRemnantMassByMullerMandel(const double, const double): Reached maximum number of iterations when calcuating remnant mass (MULLERMANDEL), (ObjectId = 10)
0: Evolution stopped because an error occurred: (Main_Sequence_>_0.7 -> Neutron_Star) + (Main_Sequence_>_0.7 -> Early_Asymptotic_Giant_Branch)

In case they're useful, here are some of the relevant parameters:
Core Mass = 2.09505
He Core Mass = 3.15706
Error thrown in first block
r = 0.014331
pBH = 0.0190097
pCompleteCollapse = 0.0158414

**Label the issue**
`urgency_low`      - This issue is not urgent
`severity_minor`    - This is a minor bug with minimal impact

**To Reproduce**
Run the following binary:
```
./COMPAS --random-seed 1725544932 -n 1 --initial-mass-min 10.0 --metallicity-distribution LOGUNIFORM --chemically-homogeneous-evolution-mode PESSIMISTIC --mode BSE
```

**Expected behavior**
No error.

**Versioning (please complete the following information):**
 - OS: [e.g. MacOS 12.4]
 - COMPAS [e.g. v03.01.07]

**Comment by ilyamandel** [comment]
There's nothing nefarious happening here, it's just that we need to pick a number from a Gaussian that's in a fairly narrow range.  Given the cheapness of this process, and the rarity of ending up in a narrow range, the simplest solution is just to increase the allowed number of iterations.  I'll do that for now.  In this example, a single extra iteration would have been enough, but I'll increase it 10-fold.  I'll also introduce a better safety fallback if we still fail to pick a suitable random number.

**Comment by ilyamandel** [comment]
Fixed in #1222 .
